src/micro_http/src/server.rs

```diff
@@ -379,6 +380,22 @@ impl HttpServer {
         Ok(parsed_requests)
     }
 
+    /// This function is responsible with flushing any remaining outgoing
+    /// requests on the server.
+    ///
+    /// Note that this function can block the thread on write, since the
+    /// operation is blocking.
+    pub fn flush_outgoing_writes(&mut self) {
+        for (_, connection) in self.connections.iter_mut() {
+            while connection.state == ClientConnectionState::AwaitingOutgoing {
```

Comment:
After the while loop, I would also set the connection state to `AwaitingIncoming`, so that if we use this method in the future in a context where we won't immediately exit the process, the state of the connection is still valid.

Reply:
From what I see, the connection state is also set to `AwaitingIncoming` on a successful write.